extension/storage/dbstorage/client.go

```diff
@@ -94,13 +98,16 @@ func (c *dbStorageClient) Batch(ctx context.Context, ops ...storage.Operation) e
 		case storage.Delete:
 			err = c.Delete(ctx, op.Key)
 		default:
+			_ = tx.Rollback()
 			return errors.New("wrong operation type")
 		}
 
 		if err != nil {
+			_ = tx.Rollback()
```

Comment:
We should probably let user know about failed rollback as well
```suggestion
			err = multierr.Append(err, tx.Rollback())
```